### PR TITLE
LKE-12414 Graph query stats

### PR DESCRIPTION
**Pull request by mzorkany**

### CODE REVIEW

File: src/api/GraphQuery/queryCRUD.ts [hunk 1]
```diff
@@ -102,3 +102,21 @@ export interface ConvertQueryResponse {
...
   dialect: GraphQueryDialect;
 }
+
+export interface GetQueryStatsParams extends IDataSourceParams {
+  query: StructuredGraphQuery;
+  dialect: GraphQueryDialect.QUERY_BUILDER;
+}
+
+export interface GetQueryStatsResponse {
+  resultCount: {
```
Comment (hsenbassam) [inline src/api/GraphQuery/queryCRUD.ts:112]: is it better to have accuracy and value in some dedicated interface ?
```
 resultCount: QueryStatsResultCount

export interface QueryStatsResultCount {
    accuracy: QueryStatsCountAccuracy;
    value: number;
}
```

File: src/api/GraphQuery/queryCRUD.ts [hunk 2]
```diff
@@ -102,3 +102,21 @@ export interface ConvertQueryResponse {
   query: string;
   dialect: GraphQueryDialect;
 }
+
+export interface GetQueryStatsParams extends IDataSourceParams {
+  query: StructuredGraphQuery;
+  dialect: GraphQueryDialect.QUERY_BUILDER;
```
Comment (hsenbassam) [inline src/api/GraphQuery/queryCRUD.ts:108]: As dialect is always queryBuilder, so it is mandatory to always pass it as param from the request ?

File: src/api/GraphQuery/queryCRUD.ts (hunk 1 above)
Comment (gmasclet) [reply src/api/GraphQuery/queryCRUD.ts:112]: I'm not sure I see the benefit of introducing this interface. In any case, we can add it later if we change our minds.

File: src/api/GraphQuery/queryCRUD.ts (hunk 2 above)
Comment (gmasclet) [reply src/api/GraphQuery/queryCRUD.ts:108]: The idea of Mahmoud was to prepare the API to support other dialect. That's debatable as we could also have considered adding the field later.

**Pull request merged**